### ISSUE TITLE
Optionally Allow Widget Groups Per Submenu

### ISSUE THREAD

**Issue by Demonseed-jpg** [jurialmunkey/skin.arctic.fuse#369]
### Skin section

Widgets

### Feature description

Hello Jurialmunkey! Hope all is well!

I am looking to be able to have the widget groups on a Complex layout be based off the submenu selection instead of the Category selection (preferably optionally choose as there may be use cases to keep it by Category for others). 

### Why is this feature needed? Please justify and explain.

Currently when designing a hub with the Complex layout, I can set up different widget groups per Category. Let’s say I have an Anime section, with Categories for Movies and Shows. Today I can have different widget groups for each one. 

As I am looking to figure out what my layout is going to be, I found myself wanting to break down the Category section by submenu. For example, I have an Anime hone menu with Categories for Movies and Shows. I found that I would like to break down movies to a Trending, Top, etc submenus, each with their own widgets groups instead of loading all the widgets I want on the single Category.  This way when highlighting Trending, I can load 2-3 widgets related to Trending Anime movies, while when I load Top, I can load 2-3 widgets related to Top anime movies. 

### Screenshots and Additional Info

I can provide screenshots if needed. Apologies if this has been brought up in another section!

### Checklist

- [X] I have made a separate issue for each feature I am requesting.
- [X] I have searched the issues and this feature has not been requested.
- [X] I am using the most recent skin version.

**Comment by jurialmunkey** [comment]
Yep, I understand what you're asking. Had a similar idea in the past but I wasn't sure if people would find it useful so I never implemented it.

Technically should be possible as static submenu items can have submenus/widgets lists just like main items (and on and on forever really).

I'm thinking rather than it being an entirely separate group, they'd go in the same main group inbetween the adapative submenu widgets and the category widgets: i.e.
-- Adapative widget
-- Widgets for submenu
-- Widgets for category


Note that it won't make any difference on loading times, since Kodi will load all the widgets in the window regardless of whether or not they are visible. It's only useful for categorising.

**Comment by Demonseed-jpg** [comment]
Well that’s awesome to hear that it’s possible, though I am not sure I fully understand your idea about putting them in the same main group. 

Behaviorally, would it be one group of widgets that each (widgets for Adaptive, Category, and Submenu) can pull from individually? So I can show widgets 1-5 for Adaptive, widgets 1 and 3 for Category, and widgets 2 and 4 for Submenu for example?

**Comment by jurialmunkey** [comment]
Example:

I have two categories "Movies" and "TV".

- "Movies"
  - Category widgets: "Random" and "Recent"
  - Submenu items: "Trending" and "Top" set to "Use as widget"
    - "Trending" submenu widgets: "Most votes" "Highest rating"
    - "Top" has no submenu widgets
- "TV"
  -  Category widgets: "Up Next" "In-Progress"

Focusing `Movies > Trending` displays these widgets:
- "Trending" (from adapative)
- "Most votes" (from submenu)
- "Highest rating" (from submenu)
- "Random" (from category)
- "Recent" (from category)

Focusing `Movies > Upcoming` displays:
- "Upcoming" (from adapative)
- "Random" (from category)
- "Recent" (from category)

Focusing `TV` displays:
- "Up Next" (from category)
- "In progress" (from category)

**Comment by Demonseed-jpg** [comment]
Gotcha, that makes sense! So it sounds like I can just only set widgets for the submenus and have them be appear completely different even tho they are part of the same group under the hood if I am understanding this right.

**Comment by Bindou01** [comment]
I find this idea very interesting.  Even if I tell myself that it would be even better suited to a horizontal global display mode, like AH2.  I support this idea, because it also offers new perspectives in how to design the interface.

**Comment by psp2111-ADSLGATE** [comment]
I cant wait for this to be implemented, it will make Arctic Fuse widgets really Fuse together and allow for next level of customization.
